tests/contrib/aiopg/test.py

```diff
@@ -198,6 +199,26 @@ def test_patch_unpatch(self):
         assert spans, spans
         assert len(spans) == 1
 
+    @run_in_subprocess(env_overrides=dict(DD_SERVICE="mysvc"))
```

Comment:
Do we need to do this as a subprocess test?

Reply:
We don't _need_ to.. it's just a bit more elegant than using testing utils. 🤷‍♂

Reply:
If we have tests that assert that `DD_SERVICE="mysvc"` -> `config.service = "mysvc", then we probably don't need to test via env variables here, we can probably just have:

```python
def test_service(self):
    with override_global_config(service="mysvc"):
        # generate spans
        spans = self.get_spans()
        assert spans[0].service != "mysvc"
```

Reply:
if the `with override_global_config():` is starting to be a huge pain, we can figure out a better way.

Maybe there is a way via pytest fixtures, or maybe we can have a function wrapper or something (e.g. `override_global_config` returns an instance of a class which defines both `__enter__`/`__exit__` as well as `__call__`, if `__call__` is used then it is treated as a decorator which uses itself as a context manager to call the wrapped function.

Reply:
I guess I just have an internal bias against `override_global_config`, not because of the ergonomics, but because I don't really trust it to perfectly emulate what happens in reality. It's a test helper that is written to perform the same side-effects as the expected real-world usage.

It's maybe a bit irrational of me..  but I like the idea of not having to depend on yet another piece of code that we wrote in order to test our functionality 😛.